File: sdks/python/examples/ollama_generate.py
#!/usr/bin/env python3
"""
Ollama generate example using MLXR Python SDK.

Demonstrates the Ollama-compatible generate API.
"""

from mlxrunner import MLXR

def main():
    client = MLXR()

    print("Ollama Generate Example")
    print("=" * 50)

    # Non-streaming generate
    print("\n1. Non-streaming generate:")
    response = client.ollama.generate(
        model="TinyLlama-1.1B",
        prompt="Why is the sky blue?",
        stream=False
    )
    print(f"Response: {response.response}")

    # Streaming generate
    print("\n2. Streaming generate:")
    print("Prompt: Write a haiku about computers\n")
    print("Response: ", end="", flush=True)

    for chunk in client.ollama.generate(
        model="TinyLlama-1.1B",
        prompt="Write a haiku about computers",
        stream=True
    ):
        if not chunk.done:
            print(chunk.response, end="", flush=True)

    print("\n")
    client.close()


if __name__ == "__main__":
    main()

File: sdks/swift/Examples/OllamaGenerate.swift
// Copyright © 2025 MLXR Development
// Example: Ollama-compatible text generation

import Foundation
import MLXR

@main
struct OllamaGenerateExample {
    static func main() async throws {
        // Create client
        let client = MLXRClient(config: .unixSocket())

        print("=== Ollama text generation ===\n")
        try await generateExample(client: client)

        print("\n=== Ollama streaming generation ===\n")
        try await streamingExample(client: client)
    }

    static func generateExample(client: MLXRClient) async throws {
        let request = OllamaGenerateRequest(
            model: "TinyLlama-1.1B",
            prompt: "Explain quantum computing in simple terms.",
            temperature: 0.8,
            numPredict: 150
        )

        let response = try await client.ollamaGenerate(request: request)

        print("Response: \(response.response)")
        print("Done: \(response.done)")

        if let evalCount = response.evalCount, let evalDuration = response.evalDuration {
            let tokensPerSec = Double(evalCount) / (Double(evalDuration) / 1_000_000_000)
            print(String(format: "Speed: %.2f tokens/sec", tokensPerSec))
        }
    }

    static func streamingExample(client: MLXRClient) async throws {
        let request = OllamaGenerateRequest(
            model: "TinyLlama-1.1B",
            prompt: "Write a short story about a robot.",
            stream: true,
            temperature: 0.9,
            numPredict: 200
        )

        print("Streaming response: ", terminator: "")

        let stream = try await client.ollamaGenerateStream(request: request)

        for try await chunk in stream {
            print(chunk.response, terminator: "")
            fflush(stdout)

            if chunk.done {
                print("\n")
                if let evalCount = chunk.evalCount, let evalDuration = chunk.evalDuration {
                    let tokensPerSec = Double(evalCount) / (Double(evalDuration) / 1_000_000_000)
                    print(String(format: "\nSpeed: %.2f tokens/sec", tokensPerSec))
                }
            }
        }
    }
}
